graphql/schemabuilder/pagination.go

```diff
@@ -393,55 +395,96 @@ func (c *connectionContext) pagesFromEdges(edges []Edge, limit int) (pages []str
 	return pages
 }
 
-func (c *connectionContext) applyTextFilter(ctx context.Context, nodes []interface{}, args PaginationArgs) ([]interface{}, error) {
-	if args.FilterText == nil || *args.FilterText == "" {
-		return nodes, nil
-	}
+func (c *connectionContext) applyBatchTextFilter(ctx context.Context, nodes []interface{}, args PaginationArgs) ([]bool, error) {
 
 	nodesToKeep := make([]bool, len(nodes))
 
 	g, ctx := errgroup.WithContext(ctx)
-	for unscopedI, unscopedNode := range nodes {
-		i, node := unscopedI, unscopedNode
-		g.Go(func() error {
-			keep := false
-			for name, filterField := range c.FilterTextFields {
-				// Resolve the graphql.Field made for sorting.
-				text, err := filterField.Resolve(ctx, node, nil, nil)
+	g.Go(func() error {
+		for name, filterField := range c.FilterTextFields {
+			// Resolve the graphql.Field made for sorting.
+			if filterField.Batch || filterField.UseBatchFunc(ctx) {
+
+				texts, err := filterField.BatchResolver(ctx, nodes, nil, nil)
 				if err != nil {
 					return err
 				}
 
-				// Only strings are allowed for FilterText fields.
-				textString, ok := text.(string)
-				if !ok {
-					return fmt.Errorf("filter %s returned %T, must be a string", name, text)
-				}
+				for i, text := range texts {
+					// Only strings are allowed for FilterText fields.
+					textString, ok := text.(string)
+					if !ok {
+						return fmt.Errorf("filter %s returned %T, must be a string", name, text)
+					}
 
-				if filter.Match(textString, *args.FilterText) {
-					keep = true
-					break
+					if filter.Match(textString, *args.FilterText) {
+						nodesToKeep[i] = true
+					}
 				}
 			}
 
-			nodesToKeep[i] = keep
+		}
+		return nil
+	})
+
+	if err := g.Wait(); err != nil {
+		return nil, err
+	}
+
+	return nodesToKeep, nil
+}
+
+func (c *connectionContext) applyTextFilter(ctx context.Context, nodes []interface{}, args PaginationArgs) ([]interface{}, error) {
+	if args.FilterText == nil || *args.FilterText == "" {
+		return nodes, nil
+	}
+
+	nodesToKeep := make([]bool, len(nodes))
+
+	g, ctx := errgroup.WithContext(ctx)
+	for name, filterField := range c.FilterTextFields {
+		if !filterField.Batch && !filterField.UseBatchFunc(ctx) {
+			for unscopedI, unscopedNode := range nodes {
+				g.Go(func() error {
+					i, node := unscopedI, unscopedNode
```

Comment:
can't do this here, we need to do this outside of the goroutine function or the `unscopedI, unscopedNode` variables will all be the same "last" parameter for all the goroutine functions (we'll all override the same point and have race conditions).

Reply:
We should probably also write a test that would catch this.  Any test that has multiple nodes should probably work.